### PR TITLE
Merge master + fix bugs

### PR DESCRIPTION
**Pull request by soywod**

### CODE REVIEW

File: src/config/model.rs [hunk 1]
```diff
@@ -26,6 +27,7 @@ pub struct Account {
     // Override
     pub name: Option<String>,
     pub downloads_dir: Option<PathBuf>,
+    pub signature_delimiter: Option<String>,
```
Comment (TornaxO7) [inline src/config/model.rs:30]: What's your intention? Do you want to be able to have multiple signatures for one account?

File: src/config/model.rs [hunk 2]
```diff
@@ -382,13 +400,21 @@ impl Config {
...
             .signature
             .as_ref()
             .or_else(|| self.signature.as_ref());
-
-        sig.and_then(|sig| fs::read_to_string(sig).ok())
+        sig.and_then(|sig| shellexpand::full(sig).ok())
+            .map(|sig| sig.to_string())
+            .and_then(|sig| fs::read_to_string(sig).ok())
             .or_else(|| sig.map(|sig| sig.to_owned()))
+            .map(|sig| String::new() + sig_delim + sig.as_ref())
```
Comment (TornaxO7) [inline src/config/model.rs]: Hm... wouldn't a `concat![sig_delim, sig.as_ref()]` suit here?

File: src/main.rs [hunk 3]
```diff
@@ -43,6 +47,18 @@ fn run() -> Result<()> {
         env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "off"),
     );
 
+    let raw_args: Vec<String> = env::args().collect();
+    if raw_args.len() > 1 && raw_args[1].starts_with("mailto:") {
```
Comment (TornaxO7) [inline src/main.rs:51]: Is this meant to be used if you press in a webbrowser on an email address?

File: src/msg/cli.rs [hunk 4]
```diff
@@ -469,6 +467,57 @@ fn msg_matches_reply(ctx: &Ctx, matches: &clap::ArgMatches) -> Result<bool> {
...
+        encoding: ContentTransferEncoding::Base64,
+        bcc: Some(bcc),
+        cc: Some(cc),
+        custom_headers: None,
+        in_reply_to: None,
+        message_id: None,
+        reply_to: None,
+        sender: None,
+        signature: ctx.config.signature(&ctx.account),
+        subject: Some(subject.into()),
```
Comment (TornaxO7) [inline src/msg/cli.rs:505]: I'd shorten this part by using the `Default` trait of `Envelope`:
```rust
let envelope = Envelope {
      from: vec![ctx.config.address(ctx.account)],
        to: vec![url.path().to_string()],
        encoding: ContentTransferEncoding::Base64,
        bcc: Some(bcc),
        cc: Some(cc),
        signature: ctx.config.signature(&ctx.account),
        subject: Some(subject.into()),
        .. Envelope::default()
   };
```
Agreed?

File: src/config/model.rs (hunk 1 above)
Comment (soywod) [reply src/config/model.rs:30]: Have a look at https://github.com/soywod/himalaya/pull/114. This option allows users to customize the signature delimiter. The default one is `--\n ` which is a (standard)[https://en.wikipedia.org/wiki/Signature_block#Standard_delimiter].

**Review by TornaxO7 — COMMENTED** [review]
@soywod Please respond to my questions please :>

File: src/config/model.rs (hunk 1 above)
Comment (TornaxO7) [reply src/config/model.rs:30]: Ah! Now I've got it. Hm... I'd like to add a little comment for that if we merge it (and if you agree as well of course). ;)

File: src/config/model.rs (hunk 2 above)
Comment (soywod) [reply src/config/model.rs]: Only literals can be passed to `concat!`. But I've found a more readable way: `sig_delim.to_owned() + sig.as_ref()`. Thanks for pointing it out!

**Comment by TornaxO7** [comment]
By the way, I'm sorry if I'm responding a little bit late nowadays. Currently I'm doing a period of practical training and also looking for an apartement, so I can only respond in the evening.

**Review by TornaxO7 — APPROVED** [review]
Looks good to me :+1:

**Comment by TornaxO7** [comment]
Should I merge now or do you want to apply some changes?

**Comment by TornaxO7** [comment]
(Would be happy if you could still answer my question above ;))

File: src/main.rs (hunk 3 above)
Comment (soywod) [reply src/main.rs:51]: Yes, see https://github.com/soywod/himalaya/issues/160 and https://github.com/soywod/himalaya/issues/162.

File: src/msg/cli.rs (hunk 4 above)
Comment (soywod) [reply src/msg/cli.rs:505]: :heavy_check_mark: Done

File: src/main.rs (hunk 3 above)
Comment (TornaxO7) [reply src/main.rs:51]: That's a neat feature! :D

File: src/config/model.rs (hunk 1 above)
Comment (soywod) [reply src/config/model.rs:30]: I added a comment on the `Config` struct.

**Comment by soywod** [comment]
I did all the changes, feel free to add more if you need.

> Should I merge now or do you want to apply some changes?

Good question. I propose to merge this one and to create another one for the fixes, what do you think?

**Comment by TornaxO7** [comment]
Agreed! I'll merge it

**Comment by soywod** [comment]
> By the way, I'm sorry if I'm responding a little bit late nowadays. Currently I'm doing a period of practical training and also looking for an apartement, so I can only respond in the evening.

Don't worry it's fine, don't forget that this is an open source project, we contribute whenever we can :smile: I mean, the project does not pay the bills (yet, I hope one day :smile:). I was also busy these days, I totally understand.

**Comment by TornaxO7** [comment]
Alright, what is still missing?

**Comment by soywod** [comment]
Thanks :tada: I will create a draft PR for the fixes.

**Comment by TornaxO7** [comment]
@soywod I'd like to keep working on this branch as well. Could you please tell me what kind of fixes you're trying to implement so I can do the other half?

**Comment by soywod** [comment]
If I refer to this https://github.com/soywod/himalaya/pull/173#pullrequestreview-721401259 I could take the 2 first (General, Read) and you the 2 last (Reply & Forward, Template)?

**Comment by TornaxO7** [comment]
Aye, we can do this!

**Pull request merged**